File: Quorum3Plugins/NetworkPlugin/src/plugins/quorum/Libraries/Network/NetworkRequest.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package plugins.quorum.Libraries.Network;

import java.net.*;
import java.io.*;


/**
 *
 * @author Patrick Daleiden
 */

public class NetworkRequest {
    public java.lang.Object me_ = null;

    public void Get(String url) throws Exception {
        final quorum.Libraries.Network.NetworkRequest quorumNetworkRequest = (quorum.Libraries.Network.NetworkRequest) me_;

        URL Url = new URL(url);
        HttpURLConnection connection = (HttpURLConnection) Url.openConnection();
        connection.setRequestMethod("GET");
        connection.setRequestProperty("Content-Type", "application/x-www-form-urlencoded");
        connection.setDoOutput(true);
        BufferedReader in = new BufferedReader(new InputStreamReader(connection.getInputStream(), "UTF-8"));
        String line;
        String response = "";
        while ((line = in.readLine()) != null) {
            response += line + '\n';
        }
        in.close();
//        quorumNetworkRequest.SetResponse(response);
    }
    
    public void Get(String url, String params) throws Exception {
        String formattedURL = url + "?" + params;
        System.out.println(formattedURL);
        Get(formattedURL);
    }
}

File: Quorum3/Library/Standard/Plugins/javascript/Libraries/Network/NetworkRequest.js
function plugins_quorum_Libraries_Network_NetworkRequest_(quorumRequest) {
    this.me_ = quorumRequest;
    
    this.Get$quorum_text = function (url) {
        console.log(url);
        this.me_.SetResponse$quorum_text("Response Get(url):\n" + url);
    };
    this.Get$quorum_text$quorum_text = function (url, params) {
        console.log(url);
        console.log(params);
        this.me_.SetResponse$quorum_text("Response Get(url, params):\n" + url + "\n" + params);
    };    
}
